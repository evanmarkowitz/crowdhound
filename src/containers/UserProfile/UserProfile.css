.user-profile {
  border: 1px solid rgb(167, 167, 167);
  width: 80%;
  height: 80vh;
  margin-left: auto;
  margin-right: auto;
  margin-top: 18vh;
  border-radius: 5px;
  background-image: linear-gradient( hsla(0, 0%, 50%, 0.047), rgba(128, 128, 128, 0.145))
}

.profile-img {
  width: 150px;
  height: 150px;
  border: gray 1px solid;
  border-radius: 50%;
  margin-left: auto;
  margin-right: auto;
  margin-top: -80px;
}

.user-profile-name {
  text-align: center;
  font-size: 20px;
  font-weight: normal;
  padding-top: 10px;
  color: rgb(240, 128, 128)

}