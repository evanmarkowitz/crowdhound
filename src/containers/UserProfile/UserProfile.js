import React from 'react';
import './UserProfile.css';
import dogFace1 from '../../images/dog-face-1.jpg';
import userImage from '../../images/bradly-cooper.jpg';
import { Link } from 'react-router-dom';
import { useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';


function UserProfile(props) {
  const id = parseInt(props.id)
  const { loading, error, data } = useQuery(gql`
    {
      user(id: ${id}) {
        id
        firstName
        lastName
        longDesc
        photos {
          sourceUrl
        }
        dogs {
          id
          name
          photos {
            sourceUrl
          }
        }
      }
    }
  `);

  if(loading) return <p>Loading....</p>;
  if(error) return <p>Error :</p>;

  const {firstName, lastName, longDesc, photos, dogs} = data.user

  const userImagePic = !photos[0] ? userImage : photos[0].sourceUrl

    const profileImageStyle = {
      backgroundImage: `url(${userImagePic})`,
      backgroundPosition: "center 1px",
      backgroundSize: "150px auto",
      backgroundRepeat: "no-repeat"
    }

    const dog = dogs.map(dog => {
      const dogImage = !dog.photos[0] ? dogFace1 : dog.photos[0].sourceUrl
      const dogImageStyle = {
        backgroundImage: `url(${dogImage})`,
        backgroundPosition: "center 1px",
        backgroundSize: "150px auto",
        backgroundRepeat: "no-repeat"
      }
      console.log(dogImage)
      return (
        <div className="user-profile-dog" key={dog.id}>
          <Link to={`/dogprofile/${dog.id}`} className="dog-profile-img" style={dogImageStyle}></Link>
          <h5 className="dog-profile-name">{dog.name}</h5>
        </div>
      )
    })

  

    return (
      <section className="user-profile">
        <section  className="profile-img" style={profileImageStyle} />
          <h5 className="user-profile-name">{firstName +" "+ lastName}</h5>
        <div className="user-profile-content-container">
            <div className="user-profile-content-info">
                <h5 className="user-about-me-title">ABOUT ME</h5>
                <p className="user-about-me-body">
                  {longDesc}
                </p>
                <h5 className="user-address-title">MY ADDRESS</h5>
                <address className="user-address-body">
                1600 Amphitheatre Parkway Mountain View,<br/>
                CA 94043
                </address>
            </div>
            <div className="user-profile-content-dogs-title">
              <h5 className="user-about-me-title">DOGS</h5>
              <div className="user-profile-dogs-container">
                  {dog}
              </div>
            </div>
        </div>
      </section>
    )
}

export default UserProfile