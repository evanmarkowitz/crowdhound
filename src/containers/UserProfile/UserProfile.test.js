import React from 'react';
import {UserProfile, GET_USER_QUERY} from './UserProfile';
import { MockedProvider } from '@apollo/react-testing'
import { shallow } from 'enzyme';
import wait from 'waait'
import renderer from 'react-test-renderer'








describe('UserProfile', () => {
  let wrapper;
  let mocks;
  beforeEach(() => {
      mocks = [
      {
        request: {
          query: GET_USER_QUERY,

        },
        result: {
          data: {
            user: { id: '1', firstName: 'Buck', lastName: 'buck', longDesc: 'longDesc', photos: {sourceUrl: ''} },
            dogs: { id: '2', name: 'buckdog', photos: {sourceUrl: ''}}
          },
        },
      },
    ];
    wrapper = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <UserProfile id={1}/>
      </MockedProvider>
    );

  })

  it('should map the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render loading state initially', async () => {
    await wait(2);
    expect(wrapper).toMatchSnapshot();
  });
})

