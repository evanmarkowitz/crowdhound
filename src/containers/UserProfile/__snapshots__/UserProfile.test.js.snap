// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should map the snapshot 1`] = `ShallowWrapper {}`;

exports[`UserProfile should render loading state initially 1`] = `ShallowWrapper {}`;
